packages/frontend/@n8n/design-system/src/v2/components/Checkbox/component-checkbox.md

```diff
@@ -0,0 +1,111 @@
+# Component specification
+
+Allows users to select one or more options from a list. The Checkbox component is ideal for binary (true/false) choices or when users need to select multiple options from a set of choices.
+
+- **Component Name:** N8nCheckbox
+- **Figma Component:** [Figma](https://www.figma.com/design/8zib7Trf2D2CHYXrEGPHkg/n8n-Design-System-V3?node-id=2121-630&m=dev)
+- **Reka UI Component:** [Checkbox](https://reka-ui.com/docs/components/checkbox)
+
+
+## Public API Definition
+
+**Props**
+
+- `id?: string` - The ID of the checkbox element
+- `label?: string` - Text label displayed next to the checkbox
+- `modelValue?: boolean` - The controlled checked state of the checkbox. Can be bind as `v-model`
+- `defaultValue?: boolean` - The checked state of the checkbox when initially rendered. Use when you do not need to control the state
+- `disabled?: boolean` - When `true`, prevents the user from interacting with the checkbox
+- `required?: boolean` - When `true`, indicates that the user must check the checkbox before submitting
+- `name?: string` - The name of the checkbox for form submission
+- `value?: string` - The value given as data when submitted with a name
+- `indeterminate?: boolean` - When `true`, displays an indeterminate state (useful for "select all" checkboxes)
+- `as?: string | Component` - Change the default rendered element for the one passed as a child, merging their props and behavior
```

Comment:
Food for thought:
I assume this `as` prop is here because reka supports it?
I wonder if we want actually to expose it in our wrapper.
I like the idea that our checkbox element will always render an element of role checkbox for accessibility.